### ISSUE TITLE
UniversalDetector vs charset.detect difference

### ISSUE THREAD

**Issue by fomcl** [chardet/chardet#296]
Hi, thanks for creating chardet! Today I used `chardet.detect` to find the encoding of about 1600 files. A bit later I got `UnicodeDecodeErrors` with 3 of those files when I tried opening them in `mode='r'` with the derived encoding. All three were mis-identified as windows-1252. I did a quick check in the terminal as shown below and I was surprised to see a different result: Mac Roman. Is there an explanation for this difference? I thought `chardet.detect` was the recommended way. I now use 
`UniversalDetector()` to get the correct result.

I'm using Python 3.11 with a recent chardet version.

```
   # Interpreter
   >>> contents = open(FILENAME, "rb").read()
   >>> chardet.detect(contents)
   {'encoding': 'Windows-1252', 'confidence': 0.7282676610947401, 'language':
   ''}

   # Terminal
   $ python -m chardet FILENAME
   FILENAME: MacRoman with confidence 0.7167379080370483
```

**Comment by fomcl** [comment]
Ok, some extra info: this file is fubar (well, all three of them actually). I don't blame chardet for not being able to understand mojibake. The difference in detection outcome is still remarkable, though. So I leave it to you whether you'd like to close this issue or not. I can ptobably provide you  with the file

**Comment by dan-blanchard** [comment]
Yeah, I have also noticed that sometimes these produce different results occasionally and it is surprising. In fact, we used to have a `hypothesis` test that checked exactly this, and we eventually commented it out because it would fail intermittently with no code changes. So I think the issue is most likely that parts of chardet aren't purely deterministic.

**Comment by dan-blanchard** [comment]
Fixed in #322
